### DISCUSSION TITLE
Concerning Rnote memory usage.

### DISCUSSION THREAD

**Discussion by andras168** [flxzt/rnote#701] [Q&A]
Hello, I don't know where should I ask this, but I think here will be fine. I really like Rnote for exercising math with a graphical tablet, and for taking notes about, well, everything. Recently, I just noticed something very odd. While I was exercising math, solving the examples (because, you know, exam is on the way), I opened btop and also KDE System Monitor and I saw that Rnote is "eating" a huge chunk of my memory. I have 16Gb RAM, but this is very concerning, Rnote just took 5Gb of that. (Around 2Gb is the normal RAM usage on KDE with all the apps running in background, which I use every day.)

Here are some pictures:
![rnote_memory_usage](https://github.com/flxzt/rnote/assets/51312507/ead03ea8-764a-41c6-8f0a-56be7332dc79)
![rnote_memory_usage_system_monitor](https://github.com/flxzt/rnote/assets/51312507/4b634b9a-382a-4ee2-ab9e-5e3656d972a6)

Is it possible that Rnote has a memory leak, or this is completely normal? I filled only 6 pages with handwritten notes and examples.
Here are my system specs if that's needed by any case.

![specs](https://github.com/flxzt/rnote/assets/51312507/cc6801ba-e86f-4a51-9382-0517c7487064)

**Comment by flxzt** [comment]
Here is good. There was a memory leak happening after closing tabs, I just pushed a fix for it (see issue #700 ). Are you experiencing this after opening/closing tabs containing large documents? If you installed rnote natively, consider trying out the git package from the aur or a manual build with the latest commit to test it.

What also might be happening: A memory usage spike when deserializing the document while opening a file. Then the system keeps the memory reserved, then it appears as if the app is constantly using that much memory from monitoring tools.
Unfortunately the currently used method is extremely memory-hungry for a short period of time, the plan is to switch to a more efficient one soon.

**Reply by andras168** [reply]
Thank you for your response. I installed it from AUR, indeed. I don't have issues with closing tabs, they are just fine. Furthermore, I'm on a continuous vertical layout. The problem is when I have a large document opened in a single tab, I get this huge memory usage, as the pictures show. 
What does "deserializing the document" actually cover?

**Reply by andras168** [reply]
It's getting worse and worse: 
![image](https://github.com/flxzt/rnote/assets/51312507/11648df0-600d-423c-8837-3b61206729b0)
